shared/util/codeql/util/Option.qll

```diff
@@ -45,3 +55,114 @@ module Option<TypeWithToString T> {
   /** Gets the given element wrapped as an `Option`. */
   Some some(T c) { result = TSome(c) }
 }
+
+/**
+ * Constructs an `Option` type that is a disjoint union of the given type and an
+ * additional singleton element, and has a `hasLocationInfo` predicate.
+ * `T` must have a `hasLocationInfo` predicate.
+ */
+module LocOption<TypeWithLocationInfo T> {
+  private module O = Option<T>;
+
+  final private class BOption = O::Option;
```

Comment:
What's with the `B` prefix? Is it meant to be shorthand for `Base` - in that case just use `Base` as a prefix - that's slightly clearer, I think. Alternatively, don't introduce final aliases at all - just use `instanceof` instead of extends, then we'll just need to repeat the 3 member methods with delegating `super` calls instead. I guess it's mostly a minor matter of style whether to go with `extends final` or `instanceof`.